rollup.config.js

```diff
@@ -5,56 +5,65 @@ import commonjs from 'rollup-plugin-commonjs'
 import babel from 'rollup-plugin-babel'
 import json from 'rollup-plugin-json'
 import flow from 'rollup-plugin-flow'
-import uglify from 'rollup-plugin-uglify'
+import { terser } from 'rollup-plugin-terser'
 import visualizer from 'rollup-plugin-visualizer'
 import sourceMaps from 'rollup-plugin-sourcemaps'
 import ignore from 'rollup-plugin-ignore'
 import pkg from './package.json'
 
 const cjs = {
-  format: 'cjs',
   exports: 'named',
+  format: 'cjs',
+  sourcemap: true,
+}
+
+const esm = {
+  format: 'esm',
+  sourcemap: true,
 }
 
 const commonPlugins = [
   flow({
     // needed for sourcemaps to be properly generated
     pretty: true,
   }),
+  sourceMaps(),
   json(),
   nodeResolve(),
-  sourceMaps(),
+  babel({
+    exclude: 'node_modules/**',
+    plugins: ['external-helpers'],
+  }),
   commonjs({
     ignoreGlobal: true,
     namedExports: {
       'react-is': ['isValidElementType'],
     },
   }),
-  babel({
-    plugins: ['external-helpers'],
-  }),
   replace({
     __DEV__: JSON.stringify(false), // disable flag indicating a Jest run
   }),
 ]
 
+const globals = { 'prop-types': 'PropTypes', react: 'React' }
+
 const configBase = {
   input: 'src/index.js',
-  globals: { 'prop-types': 'PropTypes', react: 'React' },
   external: ['react', 'prop-types'].concat(
     Object.keys(pkg.dependencies),
     Object.keys(pkg.peerDependencies)
```

Comment:
Safer external can be achieved with this function
```js
// \0 is rollup convention for generated in memory modules
const external = id => !id.startsWith('\0') && !id.startsWith('.') && !id.startsWith('/');
```

Reply:
Hmm would this let `devDependencies` through?

Reply:
Nope. Is it requirement?

Reply:
Need to double-check. This rollup file is a little hairy as you've seen 😵